modules/ethereum/src/lib.rs

```diff
@@ -359,19 +361,19 @@ decl_storage! {
 		/// Oldest unpruned block(s) number.
 		OldestUnprunedBlock: u64;
 		/// Map of imported headers by hash.
-		Headers: map hasher(blake2_256) H256 => Option<StoredHeader<T::AccountId>>;
+		Headers: map hasher(blake2_128_concat) H256 => Option<StoredHeader<T::AccountId>>;
```

Comment:
that can probably be `identity`, no?

Reply:
Yeah, you're probably right - good point :) I've also changed `ValidatorsSets` and `ValidatorsSetsRc`, because keys there are `A (slowly) incrementing index` (c).